src/main/java/core/basesyntax/strategy/handler/impl/OperationStrategyImpl.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.strategy.handler.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.handler.OperationHandler;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationHandlerMap) {
+        this.operationHandlerMap = operationHandlerMap;
+    }
+
+    @Override
+    public OperationHandler getHandler(FruitTransaction.Operation operation) {
+        return operationHandlerMap.get(operation);
+    }
```

Comment:
I think you don't need this class at all, but it's up to you

Reply:
agree with @Vt0rtikV